jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletHolder.java

```diff
@@ -1304,9 +1301,7 @@ protected Servlet newInstance() throws ServletException, IllegalAccessException,
         try
         {
             ServletContext ctx = getServletHandler().getServletContext();
-            if (ctx instanceof ServletContextHandler.Context)
-                return ((ServletContextHandler.Context)ctx).createServlet(getHeldClass());
-            return getHeldClass().getDeclaredConstructor().newInstance();
+            return ctx.createServlet(getHeldClass());
         }
```

Comment:
The context can be null, in which case we need to just call the constructor for the class directly.